src/iperf_api.c

```diff
@@ -1545,11 +1551,17 @@ int test_is_authorized(struct iperf_test *test){
         int ret = check_authentication(username, password, ts, test->server_authorized_users);
         if (ret == 0){
             iperf_printf(test, report_authetication_successed, username, ts);
+            free(username);
+            free(password);
             return 0;
         } else {
             iperf_printf(test, report_authetication_failed, username, ts);
+            free(username);
+            free(password);
             return -1;
         }
+        free(username);
```

Comment:
I think that is not needed (lines 1563 and 1564).. redundant to the "else"

Reply:
I agree @ralcini:  Lines 1563 and 1564 will never be reached.